src/sentry/search/events/datasets/profile_functions_metrics.py

```diff
@@ -66,6 +66,7 @@ def _resolve_cpm(
         alias: str | None,
         extra_condition: Function | None = None,
     ) -> SelectType:
+        assert self.builder.params.end is not None and self.builder.params.start is not None
```

Comment:
if these are required to be non-None -- consider updating the class that they come from to enforce that instead of requiring assertions here